chromium_src/services/network/restricted_cookie_manager.cc

```diff
@@ -5,24 +5,44 @@
 
 #include "services/network/restricted_cookie_manager.h"
 
+#include "base/feature_list.h"
+#include "components/content_settings/core/common/cookie_settings_base.h"
 #include "net/base/features.h"
 #include "net/cookies/cookie_monster.h"
+#include "net/cookies/site_for_cookies.h"
+#include "services/network/cookie_settings.h"
+#include "url/gurl.h"
+#include "url/origin.h"
 
 namespace {
 
-bool ShouldUseEphemeralStorage(const GURL& url,
-                               const url::Origin& top_frame_origin) {
+bool ShouldUseEphemeralStorage(
+    const GURL& url,
+    const url::Origin& top_frame_origin,
+    const net::SiteForCookies& site_for_cookies,
+    const network::CookieSettings* const cookie_settings) {
   if (!base::FeatureList::IsEnabled(net::features::kBraveEphemeralStorage))
     return false;
   if (url::Origin::Create(url) == top_frame_origin)
     return false;
-  return true;
+
+  bool block_3p = !cookie_settings->IsCookieAccessAllowed(
+      url, site_for_cookies.RepresentativeUrl(), top_frame_origin);
+  bool block_1p = !cookie_settings->IsCookieAccessAllowed(
+      url, url, url::Origin::Create(url));
+
+  // only use ephemeral storage for block 3p
+  if (block_3p && !block_1p)
+    return true;
+
+  return false;
```

Comment:
This could use the same simplification that I mentioned above.

Reply:
I left a TODO for this https://github.com/brave/brave-core/pull/7647/files#diff-72c915ffc3846c8565c42e47757c71a935649256894742e46a00ecd23f1ca5cdR147